src/_P145_MQxxx.ino

```diff
@@ -0,0 +1,281 @@
+      addHtml(F("<TR><TD>Humidity Value:<TD>"));
+      addTaskValueSelect(F("plugin_145_humidity_value"), P145_PCONFIG_HUM_VAL, P145_PCONFIG_HUM_TASK);
+      LoadTaskSettings(event->TaskIndex); // we need to restore our original taskvalues!
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      P145_PCONFIG_SENSORT   = getFormItemInt(F("plugin_145_sensor"));
+      P145_PCONFIG_RLOAD     = getFormItemFloat(F("plugin_145_RLOAD"));
+      P145_PCONFIG_RZERO     = getFormItemFloat(F("plugin_145_RZERO"));
+      P145_PCONFIG_REF       = getFormItemFloat(F("plugin_145_REFLEVEL"));
+      bool compensate        = isFormItemChecked(F("plugin_145_enable_compensation") );
+      bool calibrate         = isFormItemChecked(F("plugin_145_enable_calibrarion") );
+      P145_PCONFIG_FLAGS     = compensate + (calibrate << 1);
+      P145_PCONFIG_TEMP_TASK = getFormItemInt(F("plugin_145_temperature_task"));
+      P145_PCONFIG_TEMP_VAL  = getFormItemInt(F("plugin_145_temperature_value"));
+      P145_PCONFIG_HUM_TASK  = getFormItemInt(F("plugin_145_humidity_task"));
+      P145_PCONFIG_HUM_VAL   = getFormItemInt(F("plugin_145_humidity_value"));
+
+      P145_data_struct *P145_data = static_cast<P145_data_struct *>(getPluginTaskData(event->TaskIndex));
+      if (P145_data != nullptr)
+      {
+        P145_data->setSensorData(P145_PCONFIG_SENSORT, P145_PCONFIG_FLAGS & 0x0001, (P145_PCONFIG_FLAGS >> 1) & 0x0001, P145_PCONFIG_RLOAD, P145_PCONFIG_RZERO, P145_PCONFIG_REF);
+        P145_data->dump();
+      }
+      success = true;
+      break;
+    }
+
+    case PLUGIN_INIT:
+    {
+      P145_data_struct *P145_data = static_cast<P145_data_struct *>(getPluginTaskData(event->TaskIndex));
+      if (P145_data == nullptr)
+      {
+        P145_data = new (std::nothrow) P145_data_struct();
+        initPluginTaskData(event->TaskIndex, P145_data);
+        if (P145_data != nullptr)
+        {
+          P145_data->plugin_init();
+          P145_data->setSensorData(P145_PCONFIG_SENSORT, P145_PCONFIG_FLAGS & 0x0001, (P145_PCONFIG_FLAGS >> 1) & 0x0001, P145_PCONFIG_RLOAD, P145_PCONFIG_RZERO, P145_PCONFIG_REF);
+          P145_data->dump();
+        }
+      }
+      success = true;
+      break;
+    }
+
+    case PLUGIN_TEN_PER_SECOND:
+    {
+      P145_data_struct *P145_data = static_cast<P145_data_struct *>(getPluginTaskData(event->TaskIndex));
+      if (P145_data != nullptr)
+      {
+        success = P145_data->plugin_ten_per_second();
+      }
+      break;
+    }
+
+    case PLUGIN_READ:
+    {
+      P145_data_struct *P145_data = static_cast<P145_data_struct *>(getPluginTaskData(event->TaskIndex));
+      if (P145_data != nullptr)
+      {
+        // we're checking a var from another task, so calculate that basevar
+        float temperature = UserVar[P145_PCONFIG_TEMP_TASK * VARS_PER_TASK + P145_PCONFIG_TEMP_VAL]; // in degrees C
```

Comment:
Idem, check to see if `P145_PCONFIG_TEMP_TASK` is a valid task index.
Especially when trying to access elements in this `UserVar` array.